src/main/java/duke/model/lists/RouteList.java

```diff
@@ -0,0 +1,128 @@
+package duke.model.lists;
+
+import duke.commons.exceptions.DukeDuplicateRouteException;
+import duke.commons.exceptions.DukeException;
+import duke.commons.exceptions.DukeRouteNotFoundException;
+import duke.model.transports.Route;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+public class RouteList implements Iterable<Route> {
+    private List<Route> list;
+
+    public RouteList() {
+        list = new ArrayList<>();
+    }
+
+    public Route get(int index) throws IndexOutOfBoundsException {
+        return list.get(index);
+    }
+
+    public int size() {
+        return list.size();
+    }
+
+    /**
+     * Returns true if the list contains an equivalent Route as the given argument.
+     */
+    public boolean contains(Route toCheck) {
+        return list.stream().anyMatch(toCheck::isSameRoute);
+    }
+
+    /**
+     * Adds a Route to the list.
+     * The Task must not already exist in the list.
+     */
+    public void add(Route toAdd) throws DukeException {
+        if (contains(toAdd)) {
+            throw new DukeDuplicateRouteException();
+        }
+        list.add(toAdd);
+    }
+
+    /**
+     * Replaces the Task {@code target} in the list with {@code editedTask}.
+     * {@code target} must exist in the list.
+     * The Task identity of {@code editedTask} must not be the same as another existing Task in the list.
+     */
+    public void setRoute(Route target, Route editedRoute) throws DukeException {
+        int index = list.indexOf(target);
+        if (index == -1) {
+            throw new DukeRouteNotFoundException();
+        }
+
+        if (!target.isSameRoute(editedRoute) && contains(editedRoute)) {
+            throw new DukeDuplicateRouteException();
+        }
+
+        list.set(index, editedRoute);
+    }
+
+    /**
+     * Removes the equivalent Task from the list.
+     * The Task must exist in the list.
+     */
+    public void remove(Route toRemove) throws DukeException {
+        if (!list.remove(toRemove)) {
+            throw new DukeRouteNotFoundException();
+        }
+    }
+
+    public Route remove(int index) throws IndexOutOfBoundsException {
+        return list.remove(index);
+    }
+
+    public boolean isEmpty() {
+        return list.isEmpty();
+    }
+
+    /**
+     * Replaces the contents of this list with {@code Tasks}.
```

Comment:
Update the java docs still Task

Reply:
Noted, we will have to also decide on the format of javadoc comments soon.